arch/arm/core/cortex_m/mpu/arm_mpu.c

```diff
@@ -49,6 +45,58 @@ static inline u8_t _get_num_regions(void)
 #endif
 }
 
+/* Include architecture-specific internal headers. */
+#if defined(CONFIG_CPU_CORTEX_M0PLUS) || \
+	defined(CONFIG_CPU_CORTEX_M3) || \
+	defined(CONFIG_CPU_CORTEX_M4) || \
+	defined(CONFIG_CPU_CORTEX_M7)
+#include <arm_mpu_v7_internal.h>
+#elif defined(CONFIG_CPU_CORTEX_M23) || \
+	defined(CONFIG_CPU_CORTEX_M33)
+#include <arm_mpu_v8_internal.h>
+#else
+#error "Unsupported ARM CPU"
+#endif
+
+static int _region_allocate_and_init(const u8_t index,
+	const struct arm_mpu_region *region_conf)
+{
+	/* Attempt to allocate new region index. */
+	if (index > (_get_num_regions() - 1)) {
+
+		/* No available MPU region index. */
+		__ASSERT(0, "Failed to allocate new MPU region %u\n", index);
+		return -EINVAL;
+	}
+
+	LOG_DBG("Program MPU region at index 0x%x", index);
+
+	/* Program region */
+	_region_init(index, region_conf);
+
+	return index;
+}
+
+/* This internal function programs an MPU region
+ * of a given configuration at a given MPU index.
+ */
+static int _mpu_configure_region(const u8_t index,
+	const struct k_mem_partition *new_region)
+{
+	struct arm_mpu_region region_conf;
+
+	LOG_DBG("Configure MPU region at index 0x%x", index);
+
+	/* Populate internal ARM MPU region configuration structure. */
+	region_conf.base = new_region->start;
+	_get_ram_region_attr_by_conf(&region_conf.attr,
```

Comment:
Sudden change of attributes to configuration which is then converted to the attributes....
I suggest consistent naming.

Reply:
Not sure exactly what you mean with "consistent naming"
However, you are correct, I don't like having "ram" as an in-fix; this function works with any attribute. So I will update it.

Reply:
(Addressed by removing "ram" from the function  name)

Reply:
On both layers you have region attribute. However the meaning is different on different layers (it includes size or not). The naming of `_get_ram_region_attr_by_conf()` suggests, that you have an configuration which is converted to attribute. I recommend renaming this function, to avoid a mess in naming. Some thing like get_final_attr to something similar.

Reply:
Alright, I see what you want. Changing this to `_k_mem_partition_attr_to_mpu_region_attr()`